### DISCUSSION TITLE
Shape of a Shape

### DISCUSSION THREAD

**Discussion by jeff-zucker** [solid-contrib/shapes#4] [Ideas]
In order to create a catalog of shapes, curated, or otherwise, we need a minimum shape to describe shapes.  What do we want to know about a shape at first glance in a catalog such that we can easily distinguish it from other shapes, know whether it might be useful, etc.?

As a start, something like this 
```turtle
a oar:Shape ;
schema:name "Shape file for VCARD data in Solid" ;
schema:url <https://github.com/angelo-v/solid-contacts-spec/blob/main/shapes/contacts-shapes.ttl> ;
oar:shapeTarget vcard:AddressBook  ;
doap:implements <https://www.w3.org/TR/shacl/> ;
schema:provider  :AngeloV  ;
schema:description "..." ;
schema:license  :MIT .
```
*Note : oar: is my vocabulary, to be proposed soonish, if existing terms work better, please advise.*

-- @jeff-zucker

**Comment by jeswr** [comment]
All of the `oar` concepts you have here exist in SHACL. They are [`sh:targetClass`](https://www.w3.org/TR/shacl/#targetClass) and [`sh:NodeShape`](https://www.w3.org/TR/shacl/#node-shapes)

**Reply by jeff-zucker** [reply]
I thought of that, but does that apply to a  shex shape?  Don't we want something that is neutral to say this is shape X and has a representation Y in SHACL and representation Z in Shex?  Particularly with targetClass, it seems we want to be able to say there are both SHACL and SHEX shapes for the targetClass.  In the shape itself, yes, for sure we need to use the shacl terms in SHACL and shex terms in SHex, but this is metadata that needs to encompass both. 

I see maybe something like this :
```turtle
[                                                                                                                            
  a oar:Shape ;                                                                                                              
  oar:targetClass vcard:AddressBook ;                                                                                        
  oar:hasShaclSchema <some-shacl-file> ;                                                                                     
  oar:hasShexSchema <some-shex-file> ;                                                                                       
  oar:hasAssociatedSchema <some-sparql-query>, <some-form> ;
  doap:implements <solid-webid-profile-spec> ;                                                 
  oar:isImplementedIn <some-client-app> ...                                                                                  
]```

**Comment by elf-pavlik** [comment]
Why `schema:url <https://github.com/angelo-v/solid-contacts-spec/blob/main/shapes/contacts-shapes.ttl>` instead of

```ttl
<https://github.com/angelo-v/solid-contacts-spec/blob/main/shapes/contacts-shapes.ttl>
  a oar:Shape ; 
  # ...
```

**Reply by jeff-zucker** [reply]
I am working with all of the kinds of products and services in Solid and there are a variety of kinds of url -

* doap:service-endpoint
* doap:homepage
* doap:repository

and similar.  So it will not always be clear which is the primary.  Additionally, this is meant to work in a connected catalog.  We don't want to necessarily go to the thing itself, rather to a record about the thing which points to the thing itself.  If the thing has a WebID or ClientID, then yes, in that document the WebID or ClientID would be the subject and should be linked from this record with a owl:sameAs or vcard:webid or similar.
